### DISCUSSION TITLE
如果想自己开发前端该如何操作呢？

### DISCUSSION THREAD

**Discussion by jamesleekko** [halo-dev/halo#2192] [Q&A] [answered]
一枚前端，但是后端小白，想搭个小博客，前端部分想自己发挥一下，请问如何把项目的前端部分分离开，使用自己开发的呢（后台管理的前端还是用项目自带的）

**Comment by JohnNiang** [comment, marked as answer]
请参考：
- https://github.com/halo-dev/halo-admin
- https://docs.halo.run/developer-guide/core/structure

**Reply by jamesleekko** [reply]
所以还是脱离不了FreeMarker用自己全新的前端代码是么，其实我就是想用halo提供接口和后台管理

**Reply by ruibaby** [reply]
@jamesleekko 我们现在有 Content API 以及对应的 js-sdk。

- https://api.halo.run
- https://github.com/halo-dev/js-sdk

可以参考：https://github.com/ruibaby/halo-theme-anatole-vue

**Reply by jamesleekko** [reply]
好的  谢谢
